consensus/src/pipeline/virtual_processor/processor.rs

```diff
@@ -1005,18 +1010,21 @@ impl VirtualStateProcessor {
     pub fn process_genesis(self: &Arc<Self>) {
         // Write the UTXO state of genesis
         self.commit_utxo_state(self.genesis.hash, UtxoDiff::default(), MuHash::new(), AcceptanceData::default());
-        // Init virtual stores
-        self.virtual_stores
-            .write()
-            .state
-            .set(Arc::new(VirtualState::from_genesis(&self.genesis, self.ghostdag_manager.ghostdag(&[self.genesis.hash]))))
-            .unwrap();
+
         // Init the virtual selected chain store
         let mut batch = WriteBatch::default();
         let mut selected_chain_write = self.selected_chain_store.write();
         selected_chain_write.init_with_pruning_point(&mut batch, self.genesis.hash).unwrap();
         self.db.write(batch).unwrap();
         drop(selected_chain_write);
+
+        // Init virtual state
+        self.commit_virtual_state(
```

Comment:
Nothing special about this change rather than reusing a single entry point (`commit_virtual_state`) for writing to virtual state. It was more important at some point of developing this PR, but now it's just a cosmetic change I decided to stay with.